### PR TITLE
Generalize linear interpolation

### PR DESCRIPTION
**Pull request by JohnAFernandez**
This creates a (very small) api for linear interpolation between two recorded positions, velocitites, and desired velocities and moves multiplayer over to that system.  Now that it is generalized, other games systems (esp. weapons fire) can now use this code to temporarily set an object's position to what it would be between frames.

In draft pending testing.

### CODE REVIEW

**Comment by JohnAFernandez** [comment]
I was finally able to run a test for this now that I have a better dev machine.  Interpolation works just like in the old system, so I'm marking as ready for review.  This is a stepping stone to getting primary fire rates higher than the frame rate.

**Comment by JohnAFernandez** [comment]
This really needs to get in before 24.2, otherwise higher weapon fire rate than framerate is literally not going to happen.

File: code/object/object.cpp [hunk 1]
```diff
@@ -2095,3 +2100,28 @@ bool obj_compare(object* left, object* right) {
 
 	return OBJ_INDEX(left) == OBJ_INDEX(right);
 }
+
+void physics_populate_snapshot(physics_snapshot& snapshot, object* objp)
```
Comment (BMagnu) [inline code/object/object.cpp:2104]: Ideally, this takes a const pointer.

**Review by BMagnu — CHANGES_REQUESTED** [review]
A very tiny constness thing, but this looks good and clean.

File: code/object/object.cpp (hunk 1 above)
Comment (JohnAFernandez) [inline code/object/object.cpp:2104]:
```suggestion
void physics_populate_snapshot(physics_snapshot& snapshot, const object* objp)
```

File: code/object/object.h [hunk 2]
```diff
@@ -387,4 +384,30 @@ void obj_render_queue_all();
...
+
+/**
+ * @brief Populate a physics snapshot directly from the info in an object
+ *
+ * @param[in,out] snapshot Destination physics snapshot
+ * @param[in]     objp The object pointer that we are pulling information from
+ *
+ * @author J Fernandez
+ */
+void physics_populate_snapshot(physics_snapshot& snapshot, object* objp);
```
Comment (JohnAFernandez) [inline code/object/object.h:398]:
```suggestion
void physics_populate_snapshot(physics_snapshot& snapshot, const object* objp);
```

**Review by BMagnu — APPROVED** [review]

**Pull request merged**